pkg/ottl/ottlfuncs/func_replace_pattern_test.go

```diff
@@ -45,15 +76,58 @@ func Test_replacePattern(t *testing.T) {
 		{
 			name:    "replace regex match (with hash function)",
 			target:  target,
-			pattern: `passwd\=[^\s]*(\s?)`,
+			pattern: `passwd\=([^\s]*)\s?`,
 			replacement: ottl.StandardStringGetter[pcommon.Value]{
 				Getter: func(context.Context, pcommon.Value) (any, error) {
-					return "passwd=*** ", nil
+					return "$1", nil
 				},
 			},
 			function: optionalArg,
 			want: func(expectedValue pcommon.Value) {
-				expectedValue.SetStr("application 0f2407f2d83337b1f757eb1754a7643ce0e8fba620bc605c54566cd6dfd838beotherarg=notsensitive key1 key2")
+				expectedValue.SetStr("application hash(sensitivedtata)otherarg=notsensitive key1 key2")
+			},
+		},
+		{
+			name:    "replace regex match (capture group without $1)",
```

Comment:
Should we update this test to not refer to the capture group in the replacement string?

If we want to test having static text in the replacement string, I think we could update the previous test to cover this.

Reply:
Changed this to have just the static text in the replacement string without a capture group.